mlflow/store/artifact/databricks_artifact_repo.py

```diff
@@ -273,7 +286,13 @@ def log_artifacts(self, local_dir, artifact_path=None):
                 artifact_subdir = posixpath.join(artifact_path, rel_path)
             for name in filenames:
                 file_path = os.path.join(dirpath, name)
-                self.log_artifact(file_path, artifact_subdir)
+                upload_future = self.thread_pool.submit(
+                    self.log_artifact, file_path, artifact_subdir
+                )
+                upload_futures.append(upload_future)
+
+        for upload_future in upload_futures:
```

Comment:
The issue of iterating through all futures for result, is that when there's multiple errors, only the first one will be raised. Consider writing a util function to capture exceptions from all futures.